tensorboard/program.py

```diff
@@ -526,6 +526,14 @@ def get_url(self):
       host = self._flags.host
       display_host = (
           '[%s]' % host if ':' in host and not host.startswith('[') else host)
+
+    # Confirm that the connection is open, otherwise change to `localhost`
+    socket.setdefaulttimeout(1)
+    try:
```

Comment:
Now that creating the URL involves an actual network request, let's avoid doing that on each call to `get_url()` and just cache the constructed URL in a property and return that if it's called another time.

Reply:
Yes, I proposed a solution in bf05d. Another solution could be to move the code into `__init__()` of `WerkzeugServer` and change `get_url()` to only the last two lines of the code. Also, why do we have the property `_auto_wildcard` if it's simply `not host`?